Makefile

```diff
@@ -153,21 +167,37 @@ generate: $(CONTROLLER_GEN) ## Generate code
 
 .PHONY: generate-go
 generate-go: $(CONTROLLER_GEN) $(CONVERSION_GEN) ## Runs Go related generate targets
+	$(MAKE) generate-go-core
+	$(MAKE) generate-go-kubeadm-bootstrap
+	$(MAKE) generate-go-kubeadm-control-plane
+
+.PHONY: generate-go-core
+generate-go-core:
 	$(CONTROLLER_GEN) \
 		object:headerFile=./hack/boilerplate/boilerplate.generatego.txt \
 		paths=./api/... \
-		paths=./bootstrap/kubeadm/api/... \
-		paths=./controlplane/kubeadm/api/... \
 		paths=./cmd/clusterctl/api/...
 	$(CONVERSION_GEN) \
 		--input-dirs=./api/v1alpha2 \
 		--output-file-base=zz_generated.conversion \
 		--go-header-file=./hack/boilerplate/boilerplate.generatego.txt
+
+.PHONY: generate-go-kubeadm-bootstrap
+generate-go-kubeadm-bootstrap: $(CONTROLLER_GEN) $(CONVERSION_GEN) ## Runs Go related generate targets for the kubeadm bootstrapper
+	$(CONTROLLER_GEN) \
+		object:headerFile=./hack/boilerplate/boilerplate.generatego.txt \
+		paths=./bootstrap/kubeadm/api/...
 	$(CONVERSION_GEN) \
 		--input-dirs=./bootstrap/kubeadm/api/v1alpha2 \
 		--output-file-base=zz_generated.conversion \
 		--go-header-file=./hack/boilerplate/boilerplate.generatego.txt
 
+.PHONY: generate-go-kubeadm-control-plane
+generate-go-kubeadm-control-plane: $(CONTROLLER_GEN) $(CONVERSION_GEN) ## Runs Go related generate targets for the kubeadm control plane
+	$(CONTROLLER_GEN) \
+		object:headerFile=./hack/boilerplate/boilerplate.generatego.txt \
+		paths=./controlplane/kubeadm/api/...
+
```

Comment:
Are you also planning on breaking out the generate-manifests target as well?

Reply:
just pushed that!